packages/types/src/types/provider/vsphere/VM.ts

```diff
@@ -20,4 +20,6 @@ export interface VSphereVM extends VSphereResource {
   disks: VSphereDisk[];
   // Concerns          []model.Concern `json:"concerns"`
   concerns: Concern[];
+  // prop added by the UI to implement narrowing (discriminated union)
+  providerType: 'vsphere';
```

Comment:
I will be even more implicit to show that this is not part of the original type (from go backend)
using a full jdoc style comment and adding specific note about ProviderVirtualMachine 
``` js
/**
 * Property added by the UI to implement narrowing (discriminated union).
 * This helps identify the type when it's part of the ProviderVirtualMachine type.
 */
```

Reply:
After the property was moved to the Resource, the usage will not be limited to ProviderVirtualMachine.